### PR TITLE
fix: add missing rhel9 ngc ci steps

### PR DESCRIPTION
**Pull request by nobbs**
Hi again 👋🏻 ,

migrating this PR from Gitlab as suggested by @ArangoGutierrez.

As you're apparently planning to migrate to the main repo from Gitlab to Github (including CICD?), this PR might be obsolete at this point. Also, as already mentioned in the original PR on Gitlab which is no longer available, I'm not quite sure if the jobs/steps defined in the `.nvidia-ci.yml` are being used anywhere, as there is no direct include in the main gitlab ci file.

If it is used to build the images for NGC, this PR _should_ be adding the images for RHEL9 if I understood the setup correctly.

As said, this might be obsolete at this point, so feel free to close this PR if it is no longer required.

### CODE REVIEW

**Comment by tariq1890** [comment]
Hi @nobbs thanks for your contribution. This MR is very much useful as we haven't migrated off of gitlab.com. Can you please rebase this on top of the latest main?

**Comment by nobbs** [comment]
Hey @tariq1890 

I've just done a rebase - the change I've also introduced here with the matrix jobs: I obviously can't test them. They should be correct, as they follow the same structure used in other places, but as I don't want to break your builds, I can also remove it from this PR/MR.

**Comment by nobbs** [comment]
> Yes, on thinking about this further - I think it's better to break this PR into two - one PR for the rhel9 release job add and another for the matrix change

done - will create the second PR once this is merged eventually.

**Review by tariq1890 — APPROVED** [review]
LGTM. Pending approval from @shivamerla

**Review by shivamerla — APPROVED** [review]

**Comment by shivamerla** [comment]
lgtm

**Pull request merged**
